scripts/survival_migration/survivalDataMigration.py

```diff
@@ -0,0 +1,196 @@
+        else:
+            print("additionalVocabularies format is incorrect, please check the example by using -h option")
+
+    # save vocabularies to file
+    separator = ','
+    oneUpdateString = separator.join(newVocAddingToOne)
+    zeroUpdateString = separator.join(newVocAddingToZero)
+    df = pd.read_csv(vocabulariesFile, sep='\t')
+
+    if len(newVocAddingToZero) > 0 or len(newVocAddingToOne) > 0:
+        # get vocabularies
+        df2 = pd.DataFrame([[oneUpdateString, zeroUpdateString]], columns=[MAPPING_TO_ONE_COLUMN, MAPPING_TO_ZERO_COLUMN])
+        df = df.append(df2, ignore_index=True)
+        if len(newVocAddingToOne) > 0:
+            print("vocabulary {} has been added to one mapping".format(oneUpdateString))
+        if len(newVocAddingToZero) > 0:
+            print("vocabulary {} has been added to zero mapping".format(zeroUpdateString))
+
+    df.to_csv(vocabulariesFile, index=False, sep='\t', header=True)
+
+def migrate_file(args):
+    clinicalFile = args.clinicalFile
+    vocabulariesFile = args.vocabulariesFile
+    shouldOverrive = args.override
+    newFile = os.path.dirname(args.clinicalFile) + "/new_data_clinical_patients.txt"
+    tmpFile = os.path.dirname(args.clinicalFile) + "/tmp_data_clinical_patients.txt"
+
+    # get vocabularies
+    vocabulariesMappingToOne = []
+    vocabulariesMappingToZero = []
+    generateVocabularies(vocabulariesMappingToOne, vocabulariesMappingToZero, vocabulariesFile)
+
+    if args.additionalVocabularies != None and args.additionalVocabularies != '':
+        splitAdditionalVocabularies(args.additionalVocabularies, vocabulariesMappingToOne, vocabulariesMappingToZero, vocabulariesFile)
+
+    print("vocabularies will be mapping to one: ")
+    print(vocabulariesMappingToOne)
+    print("vocabularies will be mapping to zero: ")
+    print(vocabulariesMappingToZero)
+
+    # write files
+    input = open(clinicalFile, 'rb')
+    output = open(newFile, 'wb')
+    tmp_output = open(tmpFile, 'wb')
+
+    writer = csv.writer(output)
+    tmpwritter = csv.writer(tmp_output)
+    for row in csv.reader(input):
+        if row[0].startswith('#'):
+            writer.writerow(row)
+        else:
+            tmpwritter.writerow(row)
+    input.close()
+    output.close()
+    tmp_output.close()
+
+    # read file and replace nan
+    df = pd.read_csv(tmpFile, sep='\t')
+    df = df.replace(np.nan, '', regex=True)
+
+    # find attributes candidates
+    candidates = list(filter(lambda column: column.endswith('_STATUS'), df.columns))
+    targetAttributes = list(filter(lambda candidate: (candidate[:candidate.index("_STATUS")] + '_MONTHS') in df.columns, candidates))
+
+    for targetAttribute in targetAttributes:
+        df[targetAttribute] = df.apply(lambda row: migrateAttibuteValue(row[targetAttribute], vocabulariesMappingToOne, vocabulariesMappingToZero), axis=1)
+
+
+    df.to_csv(newFile, mode='a', index=False, sep='\t', header=True)
```

Comment:
Even if I don't have the survival related attributes, the output is written to a new file. Write both comments and dataframe to output file only if length of `targetAttribute` >=1? else print a message?

Reply:
The new line character seems to be stripped from each line? The header and data records are written to single line (when opened in textwrangler. Seems to be fine in excel or textedit). See the snippet below. 
![Screen Shot 2020-05-11 at 11 45 48 AM](https://user-images.githubusercontent.com/34350829/81581767-2dfd6e00-937d-11ea-86ce-1402f924660a.png)

Reply:
Seems to be my textwrangler issue. Using 4.5.12. Let me upgrade and try again.